### PR TITLE
Further updates evac shelters, adding vandalized and used variants.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds vandalized and used variants to evac shelters and tweaks new item spawns"```

#### Purpose of change
the evac shelters were built months before the cataclysm and left underwatched.  While they are meant to be a bit out of the way and it hasn't been *that* long, some of them have doubtless been vandalized.

#### Describe the solution
This establishes a nested mapgen framework that can be used to add other variations to the fillings of evac shelters. I made the default "pristine" center and a vandalized variant with broken windows and graffiti everywhere, and very few supplies. Amusingly enough it probably still has more supplies than the original.

I also added another variant, now the most common one: the "used" shelter. This one has a random selection of supplies, with the best stuff picked over, but still plenty of useful things here and there. As well, it has graffiti from other survivors that have come and gone, presumably taken away on FEMA buses to their likely deaths in the FEMA camps. I think this will add a lot more lore to the first chunk of the game, and establishes the world more. It also slightly reduces the power boost from my first update: you still get plenty of stuff in the used shelters, but it's a little less monty haul if you don't roll the lucky pristine start.

While in there I also shifted the walls of shelter_2 to make it easier to build a faction camp in there since that's ostensibly what I'm trying to do in this long digression. As well, I moved things around in the basement and dropped the chance of finding extremely handy loot laying around on the floor. I've also slightly adjusted item spawns to make sure the actually useful stuff is pretty rare.

This also adds a 'toxin' to the protein bars that currently just complains in your status menu if you eat them for too long. I'd like to attach a morale penalty to it, see [this issue](https://github.com/CleverRaven/Cataclysm-DDA/issues/35882)

#### Describe alternatives you've considered
My way is the only way. Do not listen to your false idols.

#### Testing
image
Vandalized shelter

image
Shelter that has had a crowd of refugees come and go before you

image
image
image

Graffiti now adds snippets of lore and adds atmosphere

image
Vandalized shelter1

image
Used shelter 1

image
What did the previous batch of refugees need with 28 of the 30 units of bleach?

image
Vandalized shelter 2, new layout

image
Used shelter 2, new layout

image
The vandalized shelter lockers aren't always empty. There can even be a little bit of useful stuff left there.

image
Lockers in the used shelters have a much lower chance of spawning the useful portable stuff, and a slightly reduced chance of spawning the rest. There's also junk and empty dishes in the spawn list now.

image
slightly modified shelter basement layout - replaced lockers with more benches.

#### Additional context
I plan to use this framework to add some NPCs that can possibly spawn already in the shelter when you arrive, and have laid claim to their own sets of gear. That will be a third PR, this is edging up on 1000 lines already.

### CODE REVIEW

File: data/json/items/comestibles/protein.json [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "id": "protein_bar_evac",
     "type": "COMESTIBLE",
     "comestible_type": "FOOD",
-    "name": { "str": "emergency protein ration", "str_pl": "emergency protein rations" },
+    "name": { "str": "protein ration", "str_pl": "protein rations" },
```
Comment (Night-Pryanik) [inline data/json/items/comestibles/protein.json]: Plural name is redundant in this case.

File: data/json/snippets/graffiti.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      { "id": "shelter_graffiti_21", "text": "<full_name> was here and still alive" },
+      { "id": "shelter_graffiti_22", "text": "Blue 52" },
+      { "id": "shelter_graffiti_23", "text": "<given_name> I no I said Id wait for u but I gotta run, find me" },
+      {
+        "id": "shelter_graffiti_24",
+        "text": "In memoriam:\n<full_name>\n<full_name>\n<full_name>\n<full_name>\n<full_name>\nI would not be alive without all of you.  I will not forget."
+      },
+      {
+        "id": "shelter_graffiti_25",
+        "text": "This is a drawing of a cartoon character smashing a zombie corpse with a sledghammer.  Beneath it is a scrawled message: \"Gotta pulp em all\""
```
Comment (Night-Pryanik) [inline data/json/snippets/graffiti.json]:
```suggestion
        "text": "This is a drawing of a cartoon character smashing a zombie corpse with a sledgehammer.  Beneath it is a scrawled message: \"Gotta pulp em all\""
```

File: data/json/snippets/graffiti.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      },
+      { "id": "shelter_graffiti_26", "text": "They get back up.  Headshots don't work." },
+      { "id": "shelter_graffiti_27", "text": "stay out of  <city>" },
```
Comment (Night-Pryanik) [inline data/json/snippets/graffiti.json]:
```suggestion
      { "id": "shelter_graffiti_27", "text": "stay out of <city>" },
```

File: data/json/items/comestibles/protein.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/items/comestibles/protein.json]:
```suggestion
    "name": { "str": "protein ration" },
```

**Comment by jbytheway** [comment]
I discovered my first vandalized shelter recently.  If the player starts in one of these, does that mean that they miss out on the usual starter quest to find the refugee center?  Is that likely to happen to new players?

**Comment by I-am-Erk** [comment]
It does mean that. I don't think it's that big a deal. I do plan to add some new options for starting quests relatively soon here.

**Pull request merged**